pkg/compiler/types.go

```diff
@@ -81,15 +84,37 @@ var typeInt = &typeDesc{
 	Check: func(comp *compiler, t *ast.Type, args []*ast.Type, base prog.IntTypeCommon) {
 		typeArgBase.Type.Check(comp, t)
 	},
+	CheckConsts: func(comp *compiler, t *ast.Type, args []*ast.Type, base prog.IntTypeCommon) {
+		if len(args) > 0 && len(args[0].Colon) != 0 {
+			begin := args[0].Value
+			end := args[0].Colon[0].Value
+			if len(args) > 1 && begin == 0 && int64(end) == -1 {
```

Comment:
This only makes sense if we have align, right?
Otherwise I think we will get division by 0 in `return begin + (r.Uint64() % (end - begin + 1))`.

Reply:
Yes. the `len(args) > 1` condition should ensure we only allow these values when we have an `align` argument... or have I missed something?

Reply:
Right.